lib/wopinator/lock.rb

```diff
@@ -0,0 +1,93 @@
+require 'time'
+require 'wopinator/lock_id'
+
+module Wopinator
+  class Lock
+    EXPIRES_IN = 1800.freeze # 30 minutes
+
+    attr_reader :id, :old_id, :timestamp
+
+    def initialize(id, old_id = nil, timestamp = Time.now)
+      self.id = id
+      self.old_id = old_id
+      self.timestamp = timestamp
+    end
+
+    def eql?(lock)
+      return false if invalid? || lock.invalid?
+
+      if old_id.empty? && lock.old_id.valid?
```

Comment:
We don't, because these are converted into LockId objects in the setters, which then handle 'nil' internally.